coq-builtin.elpi

```diff
@@ -910,6 +910,16 @@ external pred coq.hints.add-mode i:gref, i:string, i:list hint-mode.
 % GR
 external pred coq.hints.modes i:gref, i:string, o:list (list hint-mode).
 
+% [coq.hints.set-opaque C DB Opaque] Like Hint Opaque C : DB (or Hint
+% Transparent, if the boolean is ff).
+% Supported attributes:
+% - @local! (default: false)
```

Comment:
I don't think we need this right now (@gstew5 can correct me if I'm wrong), but wouldn't you eventually want to support `#[export]` locality as well for all hints? Would supporting that break the API?

Reply:
Actually, IIUC this line of code:
```
let locality = if options.local = Some true then Hints.Local else Hints.Export in
```

We're likely to need `#[global]` as well, at least since sometimes we sometimes use modules as namespaces.

More in general, `#[export] Hint Opaque` is reasonable but not as a default: a reasonable proof might spawn typeclass searches for modules that are required but not imported, but that shouldn't forget their opacity settings.

Reply:
My understanding was that export was the (future) default, and global was deprecated. I can add global if you need it, but if I got it right I won't make it the default. @ppedrot

Reply:
"Not default" sounds very acceptable :-).

Reply:
@ppedrot could you confirm that:
- default: export
- optional: local
- deprecated: global

is future proof?

Reply:
Yes. You can still consider that `global` is not deprecated, it's just never going to be the default anymore.